app/models/Photo.java

```diff
@@ -0,0 +1,165 @@
+package models;
+
+
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.fasterxml.jackson.annotation.JsonView;
+import org.apache.commons.io.FilenameUtils;
+import play.Logger;
+import com.avaje.ebean.Model;
+import play.mvc.Http;
+import plugins.S3Plugin;
+import utilities.PersistenceManager;
+import utilities.Validation;
+import utilities.View;
+
+import javax.persistence.*;
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.UUID;
+
+
+
+/**
+ * Created by Ejub on 18/03/16.
+ */
+
+
+@Entity
+@Table(name = "abh_photo")
+public class Photo extends Model implements Validation {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @JsonView(View.BasicDetails.class)
+    private long photoId;
+    @JsonView(View.AdditionalDetails.class)
+    private String bucket;
+    @JsonView(View.AllDetails.class)
+    private String name;
+    @JsonView(View.AllDetails.class)
+    private String sizeType;
+    @Transient
+    @JsonView(View.AdditionalDetails.class)
+    private File file;
+    @JsonView(View.AllDetails.class)
+    private boolean isDefault;
+    @JsonView(View.AllDetails.class)
+    @ManyToOne(cascade = CascadeType.ALL)
+    private Restaurant restaurant;
+
+    public Photo(Http.MultipartFormData.FilePart upload, int restaurantId) {
+        sizeType = "original";
+        String fileExtension = FilenameUtils.getExtension(upload.getFilename());
```

Comment:
extension, file size and mime type should be extracted from file's metadata, if available, and if not, that request data should be used (Content-Length header and file extension from the name)

you should check out **Apache Tika** for extracting these data (https://tika.apache.org/1.12/examples.html) providing input stream, not fully loaded file